jcl/src/java.base/share/classes/openj9/internal/foreign/abi/InternalDowncallHandler.java

```diff
@@ -292,13 +290,20 @@ private void validateMemScope(ResourceScope memScope) throws IllegalStateExcepti
 	}
 	/*[ENDIF] JAVA_SPEC_VERSION >= 17 */
 
-	/*[IF JAVA_SPEC_VERSION <= 19]*/
+	/*[IF JAVA_SPEC_VERSION >= 20]*/
+	/* Intended for memSegmtOfPtrToLongArgFilter that converts the memory segment
+	 * of the passed-in pointer argument to long.
+	 */
+	private final long memSegmtOfPtrToLongArg(MemorySegment argValue) throws IllegalStateException {
+		addMemArgScope(argValue.scope());
+		return UpcallMHMetaData.getNativeArgRetSegmentOfPtr(argValue);
+	}
+	/*[ELSEIF JAVA_SPEC_VERSION >= 18] */
 	/* Intended for memAddrToLongArgFilter that converts the memory address to long.
 	 * Note: the passed-in argument can be an instance of MemoryAddress, MemorySegment
 	 * or VaList which extends Addressable in OpenJDK since Java 18 featured with
 	 * JEP419 (Second Incubator).
 	 */
```

Comment:
This was a good opportunity to remove dead code: we don't need to be concerned with Java 18 or Java 19 here anymore.

Reply:
I will need to go over all code related to JDK18/19 to see what needs to be removed.

Reply:
I will clean up the outdated code via https://github.com/eclipse-openj9/openj9/pull/17352 (not yet finished) and need to verify the existing code to ensure it works good on JDK17/20.